libkbfs/folder_branch_ops.go

```diff
@@ -4593,6 +4619,63 @@ func (fbo *folderBranchOps) unstageAfterFailedResolution(ctx context.Context,
 	return fbo.unstageLocked(ctx, lState)
 }
 
+func (fbo *folderBranchOps) journalBranchChange(newBID BranchID) {
+	if newBID == NullBranchID {
+		return
+	}
+
+	ctx, cancelFunc := fbo.newCtxWithFBOID()
+	defer cancelFunc()
+
+	lState := makeFBOLockState()
+	fbo.mdWriterLock.Lock(lState)
+	defer fbo.mdWriterLock.Unlock(lState)
+
+	fbo.log.CDebugf(ctx, "Journal branch change: %s", newBID)
+
+	if !fbo.isMasterBranchLocked(lState) {
+		if fbo.bid == newBID {
+			fbo.log.CDebugf(ctx, "Already on branch %s", newBID)
+			return
+		}
+		panic(fmt.Sprintf("Cannot switch to branch %s while on branch %s",
+			newBID, fbo.bid))
+	}
+
+	md, err := fbo.config.MDOps().GetUnmergedForTLF(ctx, fbo.id(), newBID)
```

Comment:
Actually it's a pain in the butt to make an `ImmutableRootMetadata` in `tlfJournal` -- it only has easy access to `ImmutableBareRootMetadata`.  It seems the easiest way to get the non-bare version without duplicating code is to go through `journalMDOps`, which is what the code here is doing anyway.  So I think I'm going to leave it as is with the branch ID for now.